src/v/net/connection.cc

```diff
@@ -40,7 +42,9 @@ bool is_reconnect_error(const std::system_error& e) {
     if (e.code().category() == ss::tls::error_category()) {
         return absl::c_any_of(
           ss_tls_reconnect_errors, [v](int ec) { return v == ec; });
-    } else {
+    } else if (
+      e.code().category() == std::system_category()
+      || e.code().category() == std::generic_category()) {
```

Comment:
i've been reading the cppreference pages, and i still don't quite understand what the difference is between these... oh well!

Reply:
The "correct" way is not to inspect the category, but to compare with the error enum:
```c++
    if (e.code().category() == ss::tls::error_category()) {
        return absl::c_any_of(
          ss_tls_reconnect_errors, [v](int ec) { return v == ec; });
    } else if (
      e.code() == std::errc::connection_refused
      || e.code() == std::errc::host_unreachable
      || e.code() == std::errc::timed_out
      ...
```
Which provides a mapping between `std::system_category` and `std::generic_category`, I can't remember which is which, but on the platforms we care about , one is a subset of the other, and that subset has the same numbers.

Reply:
cool thx